libvast/vast/bitmap_algorithms.hpp

```diff
@@ -466,4 +468,40 @@ auto all(const Bitmap& bm) {
   return !any<!Bit>(bm);
 }
 
+/// Traverses an ID sequence in conjunction with an arbitrary range that
+/// represents blocks of IDs.
+/// @param ids The ID sequence to *select*.
+/// @param begin An iterator to the beginning of the other range.
+/// @param end An iterator to the end of the other range.
+/// @param f A function that transforms *begin* into a half-open interval of IDs
+///          *[x, y)* where *x* is the first and *y* one past the last ID.
+/// @param g A function the performs a user-defined action if the current range
+///          values falls into *(x, y)*, where `(x, y) = f(*begin)` .
+template <class IDs, class Iterator, class F, class G>
+caf::error traverse(const IDs& ids, Iterator begin, Iterator end, F f, G g) {
```

Comment:
Imho `traverse` is far to generic here, how about `select_id_ranges()`

Reply:
Definitely better. I still haven't fully wrapped my head around all the concept here, but at least no objections *at the moment.